cookbooks/aws-parallelcluster-slurm/files/default/head_node_slurm/slurm/pcluster_slurm_config_generator.py

```diff
@@ -163,21 +164,23 @@ def render_as_nodename(self):
         """Launch the rendering process."""
         config = ""
         if self.static_nodes > 0:
-            config += f"NodeName={self._static_node_name()}{self._definitions()}\n"
+            config += f"NodeName={self._static_node_name()}{self._definitions()}{self._custom_settings()}\n"
```

Comment:
Yes.
Each field is responsible to provide its own piece of text, space included.
Otherwise you should have to add logic here to check for empty strings and add spaces only when needed.